code/modules/core_implant/cruciform/rituals/machinery.dm

```diff
@@ -74,6 +74,73 @@
 	forge.produce()
 	return TRUE
 
+//Airlocks
+
+/datum/ritual/cruciform/machines/lock_door
+	name = "Activate door"
+	phrase = "Inlaqueatus"
+	desc = "Commands nearby door to be locked or unlocked."
+
+/datum/ritual/cruciform/machines/lock_door/perform(mob/living/carbon/human/user, obj/item/implant/core_implant/C)
+	var/list/O = get_front(user)
+
+	var/obj/machinery/door/holy/door = locate(/obj/machinery/door/holy) in O
+
+	if(!door)
+		fail("You fail to find a compatible door here.", user, C)
+		return FALSE
+
+	if(door.stat & (BROKEN))
+		fail("[door] is off.", user, C)
+		return FALSE
+
+	door.locked ? door.unlock() : door.lock()
+	return TRUE
+
+/datum/ritual/cruciform/machines/repair_door
+	name = "Repair door"
+	phrase = "Redde quod periit"
+	desc = "Repairs nearby door at the cost of biomatter."
+
+/datum/ritual/cruciform/machines/repair_door/perform(mob/living/carbon/human/user, obj/item/implant/core_implant/C)
+	var/list/O = get_front(user)
+
+	var/obj/machinery/door/holy/door = locate(/obj/machinery/door/holy) in O
+
+	if(!door)
+		fail("You fail to find a compatible door here.", user, C)
+		return FALSE
+
+	if(door.health == door.maxhealth)
+		fail("This door doesn't need repair.", user, C)
+		return FALSE
+
+	var/turf/target_turf = get_step(user,user.dir)
+	var/turf/user_turf = get_turf(user)
+	var/obj/item/stack/material/biomatter/consumable = locate(/obj/item/stack/material/biomatter) in target_turf.contents
```

Comment:
If the person has two stacks of biomatter in their inventory, one of 50 and one of 2, there is a risk it fails if the locate grabs the small stack so you could iterate over biomatter in target_turf.contents to grab the biggest stack